foundationdb/examples/multi_version.rs

```diff
@@ -0,0 +1,87 @@
+use byteorder::ByteOrder;
+use foundationdb::api::FdbApiBuilder;
+use foundationdb::options::NetworkOption;
+use foundationdb::tuple::Subspace;
+use foundationdb::{options, Database, FdbError, Transaction};
+
+/// This example demonstrate usage of multi_version compatibility client.
+///
+/// While you still need to compile the crate with a specific FoundationDB library version,
+/// it allows you to connect to a cluster with a different API version. Be aware that using
+/// this feature might lead to divergent behaviors.
+///
+/// Ref: https://apple.github.io/foundationdb/api-general.html#multi-version-client-api
+
+const NETWORK_OPTION_EXTERNAL_CLIENT_DIRECTORY: &str = "FDB_NETWORK_OPTION_EXTERNAL_CLIENT_DIRECTORY";
+
+#[tokio::main]
+async fn main() {
+    let mut network_builder = FdbApiBuilder::default().build().expect("Failed to build API");
+    // You can either use FoundationDB network option through environment variables
+    // or through network options in code.
+    // directory specified should contain at least one libfdb.so
+    if !std::env::var(NETWORK_OPTION_EXTERNAL_CLIENT_DIRECTORY).is_ok() {
+        network_builder = network_builder
+            .set_option(NetworkOption::ExternalClientDirectory("/usr/lib/foundationdb/".to_string()))
+            .expect("Failed to add external library directory");
+    }
+    let _guard = unsafe { 
+        network_builder.boot()
+    };
+
+    // You can replace `None` with an `Option` with the path to your `fdb.cluster` file
+    let db = Database::new_compat(None)
+        .await
+        .expect("failed to get database");
+
+    let counter_key = Subspace::all().subspace(&"stats").pack(&"my_counter");
+
+	// used to catch the first cluster_version_changed error when using external clients
+	// when using external clients, it will throw cluster_version_changed for the first time establish the connection to
+	// the cluster. Thus, we catch it by doing a get version request to establish the connection
+	// The 3000ms timeout is a guard to avoid waiting forever when the cli cannot talk to any coordinators
+    loop {
+        let trx = db.create_trx().expect("could not create transaction");
+        trx.set_option(options::TransactionOption::Timeout(3000)).expect("Failed to set timeout option");
+        if let Err(e) = trx.get_read_version().await {
+            // 1039: cluster_version_changed
+            if e.code() == 1039 {
+                continue;
+            }
+
+            panic!("Unexpected error {} while initializing the multiversion client", e.code());
+        }
+        break;
+    }
+    // write initial value
```

Comment:
I think we can just do a set/get for the test

Reply:
I left increment to see value changes over multiple successful run, as its use case is to try multiple version of fdb. Can still revert to set/get of int if needed